### ISSUE TITLE
generate_distributions() Methode implementieren

### ISSUE THREAD

**Issue by grgprajal** [Cortys/SEP-Probly-WS2526#241]
### Update Mon 1st Dec - generate_distributions() 
**Branch:** `generate-distributions`  

**Plan:**
```python
def generate_distributions(self) -> dict[str, dict]:
    """
    Extracts posterior distributions (μ and ρ) from all BayesLinear layers.
    Returns a dictionary compatible with save_distributions() / load_distributions().
    """
    distributions = {}
    for name, param in self.model.named_parameters():
        if "_mu" in name:
            base_name = name.replace("_mu", "")
            distributions.setdefault(base_name, {})
            distributions[base_name]["mu"] = param.detach().cpu().clone()
        elif "_rho" in name:
            base_name = name.replace("_rho", "")
            distributions[base_name]["rho"] = param.detach().cpu().clone()
    return distributions

Nov 23
Methode generate_distributions(self) -> dict implementieren
Dictionary mit Wahrscheinlichkeitsverteilungen über alle Variablen / Relationen zurückgeben
Platzhalter-Logik reicht erstmal (kann später durch echtes Sampling ersetzt werden)
Rückgabetyp: dict[str, dict] oder ähnlich (beliebig wählbar, aber dokumentiert)
Abnahmekriterien: Aufruf liefert valides Dictionary ohne Exception
